src/mimeparser.rs

```diff
@@ -220,7 +220,8 @@ impl MimeMessage {
         let mut mail_raw = Vec::new();
         let mut gossiped_addr = Default::default();
         let mut from_is_signed = false;
-        let mut decryption_info = create_decryption_info(context, &mail, message_time).await?;
+        let mut decryption_info = prepare_decryption(context, &mail, &from, message_time).await?;
```

Comment:
I tried to fix this by never allowing future timestamps in mime_parser, either (see 525e5b0b), but this made the tests fail because now the test accounts often don't accept that the other account lent some seconds from the future and and instead assign the same current time to every message. And therefore they don't accept keychanges, since the conditions are `if message_time > self.last_seen`, not `if message_time >= self.last_seen`.

I _think_ that the correct fix here is changing all these `>` to `>=`, but that's a larger semantical change that we may want to put into another PR. Also, maybe not, at least someone wrote this code the way it is.

We can also leave the code as-is, since currently we only look at the `message_time` when _setting_ dkim_works, and when _getting_ dkim_works (which is where it would be interesting for an attacker to fake something), we use `time()` and ignore the `message_time`. (btw, when looking at the code for this, I noticed another bug, which I fixed in 9fbbe33e)

We _could_ also only let `handle_authres()` use the not-in-the-future timestamp and not modify the timestamp for the Autocrypt code, but this seems weirdly asymmetric, and also, a timestamp far in the future would do about as much harm in the Autocrypt code as in the Authres code: In the Autocrypt code, it would prevent future keychanges, since `last_seen` would become very big and the condition `if message_time > self.last_seen` would always fail.